include/checkversion.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * @file include/checkversion.php
+ *
+ * @brief save Friendica upstream version to the DB
+ **/
+
+use Friendica\Core\Config;
+
+/**
+ * @brief check the git repository VERSION file and save the version to the DB
+ *
+ * Checking the upstream version is optional (opt-in) and can be done to either
+ * the master or the develop branch in the repository.
+ */
+function checkversion_run () {
+	global $a;
+
+	logger('checkversion: start');
+
+	$checkurl = Config::get('system', 'check_new_version_url', 'none');
+
+	// check for new versions at all?
+	if ($checkurl == 'none' ) {
+		return;
+	}
+	$checkurl = "https://raw.githubusercontent.com/friendica/friendica/".$checkurl."/VERSION";
```

Comment:
This is better, but you still are trusting the user will provide exactly the right part of the version URL. A betterer version would be the following:
````php
switch ($checkurl) {
  case 'master': $checked_url = 'https://raw.githubusercontent.com/friendica/friendica/master/VERSION'; break;
  case 'develop': $checked_url = 'https://raw.githubusercontent.com/friendica/friendica/develop/VERSION'; break;
  default: $checked_url = false;
}
````
The default part is important because, again, someone could set the config value to something that isn't `master` or `develop`, but that isn't `none` either, and you want to be able to handle those cases as well.

In general, checking for the exact data you're expecting and discarding all the rest is the best way of avoiding out of bounds bugs.

Reply:
Indeed, it replaces the `if ($checkurl == 'none' )` check.

Reply:
Or I add the `return` to the default set. That would eliminate the first if-selection and avoid another.

Reply:
Yes, but then the log line "checkversion: end" would not be reached. Depending on what output you are expecting, it isn't always good to return early. Maybe just add a log message "wrong option value" to enable debug.